### PR TITLE
tests: Update extension enabling

### PR DESCRIPTION
**Pull request by spencer-lunarg**
remove `m_device_extension_names` to `AddRequiredExtensions`/`AddOptionalExtensions` system for a lot more of tests

### CODE REVIEW

File: tests/vklayertests_wsi.cpp [hunk 1]
```diff
@@ -552,11 +552,6 @@ TEST_F(VkLayerTest, SwapchainAcquireImageNoSync2KHR) {
     TEST_DESCRIPTION("Test vkAcquireNextImage2KHR with VK_NULL_HANDLE semaphore and fence");
     SetTargetApiVersion(VK_API_VERSION_1_1);
 
-    bool extension_dependency_satisfied = false;
```
Comment (jeremyg-lunarg) [inline tests/vklayertests_wsi.cpp:555]: this part (and more) is also in https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5005

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: tests/vklayertests_wsi.cpp (hunk 1 above)
Comment (spencer-lunarg) [reply tests/vklayertests_wsi.cpp:555]: Some of these I feel the extensions were being enabled and had no real reason to be enabled and mostly likely just left there from some previous change (or some copy and pasting when the test was made)

Comment (jeremyg-lunarg) [reply tests/vklayertests_wsi.cpp:555]: I don't know if I removed all the unneeded extensions but I did convert them to use the new API. Either way, one of us will end up rebasing.

**Pull request merged**